packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/reducer/index.js

```diff
@@ -64,9 +64,11 @@ export function reducer(state = initialState, action) {
           };
         }
 
+        const newTempKey = getMaxTempKey(draft.clientState.currentWorkflow.data.stages);
+
         draft.clientState.currentWorkflow.data.stages.push({
           ...payload,
-          __temp_key__: (currentWorkflow.data?.stages?.length ?? 0) + 1,
+          __temp_key__: newTempKey,
```

Comment:
Just a thought: `__temp_key__` is used to display the order? Wouldn't it be possible to rely on the index of elements in the array? 🤔

Reply:
Interesting, thanks for the explanation 👍